src/Optimizely/DecisionService/DecisionService.php

```diff
@@ -148,6 +150,152 @@ public function getVariation(Experiment $experiment, $userId, $attributes = null
+   * @param  string      $userId      user id
+   * @param  array       $attributes  user attributes
+   * @return array/null  {"experiment" : Experiment, "variation": Variation } / null
+   */
+  public function getVariationForFeatureExperiment(FeatureFlag $featureFlag, $userId, $attributes){
+
+    $feature_flag_key = $featureFlag->getKey();
+    $experimentIds = $featureFlag->getExperimentIds();
+    //Check if there are any experiment ids inside feature flag
+    if(empty($experimentIds))
+    {
+      $this->_logger->log(Logger::DEBUG,
+        "The feature flag '{$feature_flag_key}' is not used in any experiments.");
+      return null;
+    }
+
+    // Evaluate each experiment id and return the first bucketed experiment variation
+    foreach($experimentIds as $experiment_id){
+      $experiment = $this->_projectConfig->getExperimentFromId($experiment_id);
+      if( $experiment == new Experiment()){
+        continue;
+      }
+
+      $variation = $this->getVariation($experiment, $userId, $attributes);
+      if($variation instanceof Variation && $variation != new Variation){
+        $this->_logger->log(Logger::INFO,
+          "The user '{$userId}' is bucketed into experiment '{$experiment->getKey()}' of feature '{$feature_flag_key}'.");
+        return array(
+          "experiment"=> $experiment,
+          "variation" => $variation
+        );
+      }
+    }
+
+    $this->_logger->log(Logger::INFO,
+      "The user '{$userId}' is not bucketed into any of the experiments on the feature '{$feature_flag_key}'.");
+
+    return null;
+  }
+
+   /**
+   * Get the variation if the user is bucketed for one of the rollouts on this feature flag
+   * @param  FeatureFlag $featureFlag The feature flag the user wants to access
+   * @param  string      $userId      user id
+   * @param  array       $attributes  user attributes
+   * @return Variation/null
+   */
+  public function  getVariationForFeatureRollout(FeatureFlag $featureFlag, $userId, $attributes){
+    $feature_flag_key = $featureFlag->getKey();
+    $rollout_id = $featureFlag->getRolloutId();
+    if(empty($rollout_id)){
+       $this->_logger->log(Logger::DEBUG,
+        "Feature flag '{$feature_flag_key}' is not part of a rollout.");
+       return null; 
+    }
+    $rollout = $this->_projectConfig->getRolloutFromId($rollout_id);
+    if($rollout == new Rollout()){
+      return null;
+    }
+
+    $rules = $rollout->getExperiments();
+    if(sizeof($rules) == 0)
+      return null;
+
+    // Evaluate all rollout rules except for last one
+    for($i=0; $i<sizeof($rules)-1; $i++){
+      $experiment = $rules[$i];
+
+      // Evaluate if user meets the audience condition of this rollout rule
+      if (!Validator::isUserInExperiment($this->_projectConfig, $experiment, $attributes)) {
+        $this->_logger->log(
+          Logger::DEBUG,
+          sprintf('User "%s" does not meet conditions to be in experiment "%s".', $userId, $experiment->getKey())
```

Comment:
msg looks wrong to me. plz chk java sdk.

Reply:
replaced